src/connection.js

```diff
@@ -68,7 +68,7 @@ function deprecateNonNumberConfigValue(optionName, value) {
 }
 
 class Connection extends EventEmitter {
-  constructor(config) {
```

Comment:
Could this be part of the `config` hash? Like `connectImmediately` or something like that?